### ISSUE TITLE
AppStream upstream metadata is missing

### ISSUE THREAD

**Issue by SaptarshiSarkar12** [fvarrui/JavaPackager#344]
I'm submitting a…

- [ ] bug report
- [ ] feature request
- [X] other

**Short description of the issue/suggestion:**
When the [**Drifty**](https://github.com/SaptarshiSarkar12/Drifty) application is packaged using JavaPackager Maven plugin via GitHub Action, it shows the below error 👇
```shell
WARNING: AppStream upstream metadata is missing, please consider creating it
```
How can I fix it?

**Do you have outputs, screenshots, demos or samples which demonstrate the problem or enhancement?**
You can check [**the GitHub Action output**](https://github.com/SaptarshiSarkar12/Drifty/actions/runs/5429189997/jobs/9874024573#step:6:130) to see the error.

**Please tell us about your environment:**

- JavaPackager version: 1.7.2
- OS version: ubuntu-22.04 (Ubuntu-latest for GitHub Action)
- JDK version: 20.0.1
- Build tool: 
	- [X] Maven 
	- [ ] Gradle

**Other Information**
You can check the [**pom.xml**](https://github.com/SaptarshiSarkar12/Drifty/blob/master/pom.xml) file to see if any changes needs to be made in that.

**Comment by SaptarshiSarkar12** [comment]
@fvarrui Please look into the issue.

**Comment by fvarrui** [comment]
Hi @SaptarshiSarkar12!

I fear you can fix it ... unless you code a patch for JavaPackager and make a PR. 

I'll take a look into it ASAP.

Anyway, it's showing a warning message, not an error. Have you tried to run the generated AppImage?

**Comment by SaptarshiSarkar12** [comment]
> Hi @SaptarshiSarkar12!
> 
> I fear you can fix it ... unless you code a patch for JavaPackager and make a PR.
> 
> I'll take a look into it ASAP.
> 
> Anyway, it's showing a warning message, not an error. Have you tried to run the generated AppImage?

@fvarrui Yes, the AppImage runs pretty well.

**Comment by SaptarshiSarkar12** [comment]
Thank you for giving your valuable time to resolve the issue!

**Comment by SaptarshiSarkar12** [comment]
@fvarrui Please check [this docs of freedesktop about metadata file](https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html#sect-Quickstart-DesktopApps).
It says to create a **`app_name.metadata.xml`** file in **`/usr/share/metainfo/`** directory before building the AppImage. I think this file can be auto-created by **JavaPackager** before  initiating the building of AppImages. So, it might be a good feature we should bring in this project. 
@fvarrui Let me know your opinion.

**Comment by fvarrui** [comment]
@SaptarshiSarkar12 Ok, I've just check this [example](https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html#qsr-app-example) and it seems that all required data is available in JP. As it's related to issue-345, and there's already a [branch for this issue](https://github.com/fvarrui/JavaPackager/tree/issue-345), changes will be made in this branch.

**Comment by fvarrui** [comment]
> No, there is some problems regarding DEB and might be RPM also. The problem is that the description shown by **Software Install** application in ubuntu (when trying to install DEB package) is not as what is provided to JavaPackager.

Sí, do you mean that metadata file has also to be provided by DEB and RPM packages?

**Comment by SaptarshiSarkar12** [comment]
> > No, there is some problems regarding DEB and might be RPM also. The problem is that the description shown by **Software Install** application in ubuntu (when trying to install DEB package) is not as what is provided to JavaPackager.
> 
> So, do you mean that metadata file has also to be provided by DEB and RPM packages?

Yeah @fvarrui, it might help. Also, the application _**icon**_ is not present when opening DEB (might be the same case with RPM) through **Software Install** application in _**Ubuntu 23.04**_.
